src/placement/pl_map_common.c

```diff
@@ -314,16 +314,8 @@ determine_valid_spares(struct pool_target *spare_tgt, struct daos_obj_md *md,
 		/* The selected spare target is up and ready */
 		l_shard->po_target = spare_tgt->ta_comp.co_id;
 		l_shard->po_fseq = f_shard->fs_fseq;
-
-		/*
-		 * Mark the shard as 'rebuilding' so that read will
-		 * skip this shard.
-		 */
-		if (f_shard->fs_status == PO_COMP_ST_DOWN ||
-		    f_shard->fs_status == PO_COMP_ST_DRAIN) {
-			l_shard->po_rebuilding = 1;
-			f_shard->fs_tgt_id = spare_tgt->ta_comp.co_id;
-		}
+		l_shard->po_rebuilding = 1;
+		f_shard->fs_tgt_id = spare_tgt->ta_comp.co_id;
```

Comment:
I don't understand the commit comment.

Is this change because remap will also happen for UP node in colocation case, and in this case, "po_rebuilding" needs be set as well?

Reply:
I think anything that needs a spare should set the spare to rebuilding, so this change looks good to me.

Reply:
yes, this is for reintegration case. yes, po_rebuilding needs to be set to avoid fetch on that shard anyway.
As Bryon said, any targets on the remap list needs to be found a spare, either available or not available.

Reply:
Won't it break the common case of remapping a DOWNOUT target to an available target? All shard should be readable in that case.

In old placement algorithm, if a shard's original target is in DOWNOUT, it'll be remapped to a UPIN target, and this shard should be readable (rebuild for the original failed target is already finished). Is this logic changed in new placement algorithm?

Reply:
Not sure If I follow here, but why does it break the remapping a downout target to an available target? 

This is to set the shard once it find an available spare target for the shard, and I only remove the original target status check, which is actually unnecessary and confusing to me, and also wrong for reintegration.